.github/workflows/all.yml

```diff
@@ -204,21 +208,45 @@ jobs:
             docker run $DOCKER_IMAGE_ID scale --help
             docker run $DOCKER_IMAGE_ID pause --help
             docker run $DOCKER_IMAGE_ID resume --help
-      - name: Publish
-        if: github.event_name != 'pull_request'
+      - name: Publish Docker CI
         run: |
-          echo "REF=${{ github.ref }}"
+          VERSION_CI="${VERSION_CI#v}"
           echo "DOCKER_IMAGE_ID=$DOCKER_IMAGE_ID"
-          # Log into registry
-          echo "${{ secrets.DOCKER_PASS }}" | docker login -u "${{ secrets.DOCKER_USER }}" --password-stdin
+          echo "VERSION_CI=$VERSION_CI"
+          # Log into GHCR
+          echo "${{ secrets.CR_PAT }}" | docker login https://ghcr.io -u ${{ github.actor }} --password-stdin
+          docker tag "$DOCKER_IMAGE_ID" "ghcr.io/$DOCKER_IMAGE_ID/ci:$VERSION_CI"
+          docker push "ghcr.io/$DOCKER_IMAGE_ID/ci:$VERSION_CI"
+
+  publish-docker:
+    runs-on: ubuntu-latest
+    needs: [build-docker, configure]
+    if: github.event_name != 'pull_request'
+    env:
+      VERSION: ${{ needs.configure.outputs.version }}
+      VERSION_CI: ${{ needs.configure.outputs.version_ci }}
+    steps:
+      - name: Configuring variables
+        run: |
           VERSION="${VERSION#v}"
+          VERSION_CI="${VERSION_CI#v}"
+          echo "REF=${{ github.ref }}"
+          echo "DOCKER_IMAGE_ID=$DOCKER_IMAGE_ID"
           echo "VERSION=$VERSION"
-          docker tag "$DOCKER_IMAGE_ID" "$DOCKER_IMAGE_ID:$VERSION"
-          docker push "$DOCKER_IMAGE_ID:$VERSION"
+          echo "VERSION_CI=$VERSION_CI"
+      - name: Publish Docker
+        run: |
+          skopeo copy --src-creds ${{ github.actor }}:${{ secrets.CR_PAT }} --dest-creds ${{ secrets.DOCKER_USER }}:${{ secrets.DOCKER_PASS }} docker://ghcr.io/$DOCKER_IMAGE_ID/ci:$VERSION_CI docker://$DOCKER_IMAGE_ID:$VERSION
+          # We also want to tag the latest stable version as latest
+          if [[ "$VERSION" != "master" ]] && [[ ! "$VERSION" =~ (RC|rc) ]]; then
+            skopeo copy --src-creds ${{ github.actor }}:${{ secrets.CR_PAT }} --dest-creds ${{ secrets.DOCKER_USER }}:${{ secrets.DOCKER_PASS }} docker://ghcr.io/$DOCKER_IMAGE_ID/ci:$VERSION_CI docker://$DOCKER_IMAGE_ID:latest
+          fi
+      - name: Publish GHCR
+        run: |
+          skopeo copy --src-creds ${{ github.actor }}:${{ secrets.CR_PAT }} --dest-creds ${{ github.actor }}:${{ secrets.CR_PAT }} docker://ghcr.io/$DOCKER_IMAGE_ID/ci:$VERSION_CI docker://ghcr.io/$DOCKER_IMAGE_ID:$VERSION
           # We also want to tag the latest stable version as latest
           if [[ "$VERSION" != "master" ]] && [[ ! "$VERSION" =~ (RC|rc) ]]; then
-            docker tag "$DOCKER_IMAGE_ID" "$DOCKER_IMAGE_ID:latest"
-            docker push "$DOCKER_IMAGE_ID:latest"
+            skopeo copy --src-creds ${{ github.actor }}:${{ secrets.CR_PAT }} --dest-creds ${{ github.actor }}:${{ secrets.CR_PAT }} docker://ghcr.io/$DOCKER_IMAGE_ID/ci:$VERSION_CI docker://ghcr.io/$DOCKER_IMAGE_ID:latest
```

Comment:
Should we join both publishing steps into one, to not repeat the `master`/`RC` logic?

Reply:
I was uncertain until publishing PR.... I agree with you and will merge it !